core/arch/arm/plat-imx/conf.mk

```diff
@@ -387,16 +387,24 @@ CFG_MMAP_REGIONS ?= 24
 
 # Almost all platforms include CAAM HW Modules, except the
 # ones forced to be disabled
-CFG_NXP_CAAM ?= n
+CFG_NXP_CAAM ?= y
```

Comment:
I would prefer to not change this file
CFG_NXP_CAAM_RUNTIME_JR is already managed in the crypto_conf.mk, so no need to change it.
CFG_CRYPTO_DRIVER must be enabled only if the CAAM driver is to be supported at runtime to replace the Crypto Software library operations supported in the CAAM Driver.
CFG_NXP_CAAM_MKVB_DRV  should be defined in the crypto_conf.mk file where all CAAM driver features are managed.

Reply:
This PR intentionally disables `CFG_NXP_CAAM_RUNTIME_JR`, to ensure that OP-TEE does not try to access the CAAM during runtime, because this will currently lead to SoC Lockups (see https://github.com/OP-TEE/optee_os/issues/3405 and the discussion in https://github.com/OP-TEE/optee_os/pull/3349).

Reply:
I still don't understand. CFG_NXP_CAAM_RUNTIME_JR is not defined if the CFG_CRYPTO_DRIVER is not equal to 'y'.
If CFG_CRYPTO_DRIVER is not enabled, CAAM driver is not installed to be used at runtime. runtime CAAM driver has not interest if the CFG_CRYPTO_DRIVER is not used.

Reply:
> I still don't understand. CFG_NXP_CAAM_RUNTIME_JR is not defined if the CFG_CRYPTO_DRIVER is not equal to 'y'.
> If CFG_CRYPTO_DRIVER is not enabled, CAAM driver is not installed to be used at runtime. runtime CAAM driver has not interest if the CFG_CRYPTO_DRIVER is not used.

Thats right, I'll remove the explicit `CFG_NXP_CAAM_RUNTIME_JR`.

Reply:
We are doing our best to satisfy (and enable) all i.MX users. It doesn't mean that we have done as reference can correspond to everybody wanted this or this options or want to disable certain parts. Hence, if you want to re-use plat-imx but with specific build options, change your build command line to enable your option.

Reply:
As I said previously  in https://github.com/OP-TEE/optee_os/pull/3349#issue-333011100, upstream OP-TEE is expected to work with an upstream kernel, which is the reason why `CFG_NXP_CAAM` was disabled in https://github.com/OP-TEE/optee_os/pull/3349. However we now have the case where we require the CAAM driver to be used for the HUK generation (and for the initial RNG Seed), but will still run into the clock conflict issue since the kernel will disable the clocks. Therefore I enabled the CAAM driver and disabled `CFG_CRYPTO_DRIVER` which disables the CAAM runtime access. This way upstream OP-TEE can use the CAAM during initialization for the HUK and RNG seed, but won't conflict with the kernel during runtime.

> We are working on the integration of the CAAM driver on referenced platform, but before we have to ensure that enabled platforms are fully validated.
Your change can break the boot of non validated platforms.

Why was `CFG_NXP_CAAM` than enabled by default before? Either way I validated this change at least on i.MX6Q and i.MX6UL, I'd be okay to only enable these platforms.

Regarding the disabled `CFG_CRYPTO_DRIVER`, @clementfaure mentioned in https://github.com/OP-TEE/optee_os/pull/3349#issuecomment-547790617 that there was a patch set for the problem. I am fully in favor of re-enabling `CFG_CRYPTO_DRIVER` for `plat-imx` once this patchset is merged in the upstream kernel.

Reply:
So you are going the same way as us. Conclusion don't touch this configuration. We will do it.
Waiting that, enable it on your side locally either thru the build command line, either by editing this file but not pushing it to upstream. Thanks.

Reply:
> Conclusion don't touch this configuration.

I don't reach the same conclusions. I'd like HUK generation to be available with the next OP-TEE release which is due some time in april. Current kernel looks to be on schedule for a end of march release, I doubt that the CAAM kernel patches can still target v5.6. Consequently the changes could be in 5.7. which is due after the OP-TEE release. The next OP-TEE release can consequently enable `CFG_CRYPTO_DRIVER`.

Reply:
Hi @Emantor,
We still believe that ```CFG_NXP_CAAM``` should stay disabled for reasons already explained by @cneveux. The CAAM has been disabled for good reasons and we prefer to wait for a proper patch in the kernel before re-enabling it.
In the mean time, I believe this configuration should be done locally to enabled the HUK generation, but not upstreamed.